File: benchmarks/oopsla25/subjects/140.c
#include <assert.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

int f140(const char* s) {
    const char* ptr = strchr(s, 'a');
    if (ptr != NULL) {
        return ptr - s;
    }
    return -1;
}

int main(int argc, char *argv[]) {
  char my_string[10240] = {0};
  if (argc == 1) {
    read(fileno(stdin), my_string, 10240);
  } else {
    int fd = open(argv[1], O_RDONLY);
    read(fd, my_string, 10240);
    close(fd);
  }
  int i = f140(my_string);
  printf("%d\n", i);
  return 0;
}

File: benchmarks/oopsla25/subjects/140.py
#!/usr/bin/env python3

import sys

def f140(s: str):
  return s.index("a")

# note: the regex [^a]*a.* is equivalent to .*a.*

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  f140(my_string)
